x-pack/plugins/aiops/public/components/change_point_detection/chart_component.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useMemo } from 'react';
+import { type TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import { useChangePontDetectionContext } from './change_point_detection_context';
+import { useDataSource } from '../../hooks/use_data_source';
+import { useAiopsAppContext } from '../../hooks/use_aiops_app_context';
+import { useTimeRangeUpdates } from '../../hooks/use_time_filter';
+
+export interface ChartComponentProps {
+  annotation: {
+    group_field: string;
+    label: string;
+    timestamp: string;
+    reason: string;
+  };
+}
+
+const fnOperationTypeMapping: Record<string, string> = {
+  min: 'min',
+  max: 'max',
+  sum: 'sum',
+  avg: 'average',
+} as const;
+
+export const ChartComponent: FC<ChartComponentProps> = React.memo(({ annotation }) => {
+  const {
+    lens: { EmbeddableComponent },
+  } = useAiopsAppContext();
+
+  const timeRange = useTimeRangeUpdates();
+  const { dataView } = useDataSource();
+  const { requestParams, bucketInterval } = useChangePontDetectionContext();
+
+  const filters = useMemo(
+    () => [
+      {
+        meta: {
+          index: dataView.id!,
+          alias: null,
+          negate: false,
+          disabled: false,
+          type: 'phrase',
+          key: requestParams.splitField,
+          params: {
+            query: annotation.group_field,
+          },
+        },
+        query: {
+          match_phrase: {
+            [requestParams.splitField]: annotation.group_field,
+          },
+        },
+        $state: {
+          store: 'appState',
+        },
+      },
+    ],
+    [dataView.id, requestParams.splitField, annotation.group_field]
+  );
+
+  // @ts-ignore
+  const attributes = useMemo<TypedLensByValueInput['attributes']>(() => {
+    return {
+      title: annotation.group_field,
+      description: '',
+      visualizationType: 'lnsXY',
+      type: 'lens',
+      references: [
+        {
+          type: 'index-pattern',
+          id: dataView.id!,
+          name: 'indexpattern-datasource-layer-2d61a885-abb0-4d4e-a5f9-c488caec3c22',
```

Comment:
Can you explain where these names and ids in this specification come from?

Reply:
I used the Lens playground to generate the attributes, you can find out more about how it's done in this guide https://github.com/elastic/kibana/tree/main/x-pack/plugins/lens#getting-started